components/brave_wallet_ui/common/async/handlers.ts

```diff
@@ -627,4 +628,28 @@ handler.on(WalletActions.setSelectedNetworkFilter.getType(), async (store: Store
   await store.dispatch(refreshTokenPriceHistory(selectedPortfolioTimeline))
 })
 
+handler.on(WalletActions.addAccount.getType(), async (store: Store, payload: AddAccountPayloadType) => {
+  const { keyringService, walletHandler } = getAPIProxy()
+  const { accounts } = getWalletState(store)
+
+  // Creates a new account by the payloads coinType
+  const result = await keyringService.addAccount(payload.accountName, payload.coin)
```

Comment:
Can we use `hasCoinTypeAccount` below to decide if we need to call `addAccount`.

Reply:
Cant use `hooks` inside of a `handler`